packages/block-library/src/template-part/edit/name-panel.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityProp } from '@wordpress/core-data';
+import { TextControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { cleanForSlug } from '@wordpress/url';
+
+export default function TemplatePartNamePanel( { postId, setAttributes } ) {
+	const [ title, setTitle ] = useEntityProp(
+		'postType',
+		'wp_template_part',
+		'title',
+		postId
+	);
+	const [ , setSlug ] = useEntityProp(
+		'postType',
+		'wp_template_part',
+		'slug',
+		postId
+	);
+	return (
+		<TextControl
+			hideLabelFromVision
+			label={ __( 'Name' ) }
+			value={ title }
+			onChange={ ( value ) => {
```

Comment:
Same question here as above.  While passing an anonymous function into an element's `onChange` seems perfectly fine, with the component wouldn't this potentially cause unnecessary re-renders?

Reply:
Only if the component is "Pure", when components are shallow and cheap to render like this, making them "Pure" and the memoization overhead can be more expensive than just rerendering.

Reply:
Ohh right.  It will update when the parent does regardless of prop change anyways.